src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+#include "tribol/common/Parameters.hpp"
+#include "tribol/mesh/MethodCouplingData.hpp"
+
+namespace tribol
+{
+
+/**
+ * @brief Facilitates transfer to submesh low-order refined mesh
+ *
+ * This class simplifies transferring 1) a variable (e.g. displacement) from a
+ * higher-order grid function on a coarse mesh to a low-order grid function on a
+ * fine mesh and 2) a variable in the dual space (e.g. force) from a low-order
+ * grid function on a fine mesh to a higher-order grid function on a coarse
+ * mesh. The coarse mesh is intended to live on a mfem::ParSubMesh level, but
+ * this is not required to use this class.
+ */
+class SubmeshLORTransfer
+{
+public:
+  /**
+   * @brief Construct a new SubmeshLORTransfer object
+   * 
+   * @param ho_submesh_fes Higher order finite element space on the coarse mesh
+   * @param lor_submesh Refined mesh
+   */
+  SubmeshLORTransfer(
+    mfem::ParFiniteElementSpace& ho_submesh_fes,
+    mfem::ParMesh& lor_submesh
+  );
+
+  /**
+   * @brief Transfers data from a higher-order grid function on a coarse mesh
+   *
+   * Data is transferred to the low-order grid function on a fine mesh that can
+   * be accessed using GetLORGridFn().
+   *
+   * @param ho_src Source higher-order grid function on the coarse mesh
+   */
+  void TransferToLORGridFn(
+    const mfem::ParGridFunction& ho_src
+  );
+
+  /**
+   * @brief Transfers data to a higher-order grid function on a coarse mesh
+   *
+   * Data must be stored in the low-order grid function on a fine mesh accessed
+   * using GetLORGridFn().
+   *
+   * @param ho_dst Destination higher-order grid function on the coarse mesh
+   */
+  void TransferFromLORGridFn(
+    mfem::ParGridFunction& ho_dst
+  ) const;
+
+  /**
+   * @brief Access the local low-order grid function on the refined mesh
+   * 
+   * @return mfem::ParGridFunction& 
+   */
+  mfem::ParGridFunction& GetLORGridFn() { return lor_gridfn_; }
+
+  /**
+   * @brief Access the local low-order grid function on the refined mesh
+   * 
+   * @return const mfem::ParGridFunction& 
+   */
+  const mfem::ParGridFunction& GetLORGridFn() const { return lor_gridfn_; }
+
+private:
+  /**
+  * @brief Create low-order grid function on the refined mesh
+  * 
+  * @param lor_mesh Refined mesh
+  * @param lor_fec Finite element collection to apply to grid function
+  * @param vdim Vector dimension of the grid function
+  * @return mfem::ParGridFunction on lor_mesh, with lor_fec and vdim specified
+  */
+  static mfem::ParGridFunction CreateLORGridFunction(
+    mfem::ParMesh& lor_mesh,
+    std::unique_ptr<mfem::FiniteElementCollection> lor_fec,
+    integer vdim
+  );
+
+  /**
+   * @brief Local low-order grid function on the refined mesh 
+   */
+  mfem::ParGridFunction lor_gridfn_;
+
+  /**
+   * @brief Low-order refined <-> higher-order coarse transfer object
+   */
+  mutable mfem::L2ProjectionGridTransfer lor_xfer_;
+};
+
+/**
+ * @brief Facilitates transferring variables from the submesh to redecomp levels
```

Comment:
See comment below at ParentRedecompTransfer class. This class (SubmeshRedecompTransfer) looked like it was the only thing we need. We create submeshes from the volume Parmesh registered with Tribol, which is independent of any LOR and redecomp. Then, if LOR, we perform the LOR transfers and obtain LOR meshes, then we send through redecomp. Then I saw the class below and got confused. Please clarify use case or how this class fits in the whole flow.

Reply:
Hopefully the updated comments cleared this up?